07-dataflow.md

```diff
@@ -1,54 +1,151 @@
- 4. The reconstructed `DST` files are suitable for analysis, but they are not accessible to users due to computing restrictions. Data must be filtered futher through a set of selections called *stripping*, which write output either in `DST` or `µDST` format.
+Let's say we could somehow store the response of every subdetector in LHCb for 
+every event, a proton-proton bunch crossing, in only one kilobyte, that is to 
+say as 8000 ones and zeroes.
+This would mean you'd need 40 gigabytes of disk space to store one second of 
+data!
+
+That's too much data for us to be able to keep all of it, the price of storage 
+is just too high, so instead we need to _filter_ the data and try to keep only 
+the events which contain something interesting.
+This raises its own problems:
+
+* How do we filter and process the recorded data quickly and accurately?
+* How do we manage all the complex tasks required to work with collision data?
+* How do we organize all the data of a single bunch crossing in a flexible way?
+* How do we configure our software flexibly without having to recompile it?
+* Can you think of more?
+
+These questions arise mostly due to two key points: the data must be processed 
+very quickly because it's arriving very quickly, and the data is complex so 
+there's a lot that can be done with it.
+
+Collisions recorded by the LHCb detector go through a specific data flow 
+designed to maximise the data-taking efficiency and data quality.
+This consists of several steps, each one being controlled by an ‘application’ 
+that processes the data event-by-event, using the data from the previous step 
+and creating the results ready for the next.
+These steps are as follows:
+
+ 1. Data from the detector are filtered through the _trigger_, which consists 
+    of the L0, implemented in hardware, and the high level trigger (HLT), 
+    implemented in software. The application responsible for the software trigger 
+    is Moore, which will be discussed in further detail in the trigger lesson.
+ 2. Triggered, raw data are reconstructed to transform the detector hits into 
+    objects such as tracks and clusters. This is done by the Brunel 
+    application. The objects are stored into an output file in a ‘DST’ format.
+ 3. The reconstructed DST files are suitable for analysis, but they are not 
+    accessible to users due to computing restrictions. Data are filtered 
+    further through a set of selections called the *stripping*, controlled by the 
+    DaVinci application which write output either in DST or ‘µDST’ (micro-DST) 
+    format. To save disk space and to speed up access for analysts, stripping 
+    selections that select similar types of decays, such as those containing fully 
+    hadronic decays of charm hadrons and those containing a dimuon pair, there a 
+    _streams_ which contain similar selections.
```

Comment:
For sure. Hows about this:

> The reconstructed DST files are suitable for analysis, but they are not 
accessible to users due to computing restrictions. Data are filtered 
further through a set of selections called the *stripping*, controlled by 
the DaVinci application which write out data either in the DST or ‘µDST’ 
(micro-DST) format. To save disk space and to speed up access for analysts, the 
output files are grouped into _streams_ which contain similar selections.
By grouping all of the fully hadronic charm selections together, for example, 
analysts interested in that type of physics don't waste time running of the 
output over the output of the dimuon selections.